### PR TITLE
Added windows executable extensions to Kubectl plugins

### LINKED ISSUE #73471: Kubectl plugin system should consider more file extensions under Windows
**What would you like to be added**:
Currently, the kubectl plugins system - when running under Windows - will determine that a plugin is executable by looking at the plugin's extension, comparing it to '.exe'.

I think we should allow plugins w/ a few more known extensions (.bat, .com, .cmd; for example).

**Why is this needed**:
This would give more of a similar experience to plugins under *nix, where shell scripts are often used.

### PR DESCRIPTION
**Pull request by dany74q**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind feature

**What this PR does / why we need it**:
Currently, the kubectl plugins system - when running under Windows - will determine that a plugin is executable
by looking at its extension, comparing it to '.exe'.

I think we should allow plugins w/ a few more known extensions (.bat, .com, .cmd; for example) -
This would give more of a similar experience to plugins under *nix, where shell scripts are often used.

**Which issue(s) this PR fixes**:

Fixes #73471

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dany74q** [comment]
CLA is now signed.

File: pkg/kubectl/cmd/plugin/plugin.go [hunk 1]
```diff
@@ -230,9 +230,13 @@ func isExecutable(fullPath string) (bool, error) {
...
 	if runtime.GOOS == "windows" {
-		if strings.HasSuffix(info.Name(), ".exe") {
-			return true, nil
+		fileExt := strings.ToLower(filepath.Ext(fullPath))
+		
+		switch fileExt {
+			case ".bat", ".cmd", ".com", ".exe":
+				return true, nil
 		}
+
```
Comment (neolit123) [inline pkg/kubectl/cmd/plugin/plugin.go]: please omit this emtpy line and squash the commits into one.

**Review by neolit123 — COMMENTED** [review]
thanks @dany74q 
LGTM added one minor nit.

/priority backlog
/ok-to-test
/remove-kind feature
/kind cleanup

**Comment by neolit123** [comment]
@dany74q 
please run `./hack/update-gofmt.sh` and commit the changes as a single commit.
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/73457/pull-kubernetes-verify/120436

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
/lgtm

looks like you have gofmt failures https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/73469/pull-kubernetes-verify/120447

**Review by soltysh — APPROVED** [review]
/approve

File: pkg/kubectl/cmd/plugin/plugin.go (hunk 1 above)
Comment (neolit123) [reply pkg/kubectl/cmd/plugin/plugin.go]: thank you @dany74q 
LGTM

/retest

**Comment by juanvallejo** [comment]
/hold

You'll also need to update https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cmd.go#L345 so that kubectl knows to look for these additional extensions on windows

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
/hold cancel
/lgtm

@dany74q Thanks for the clarification

File: pkg/kubectl/cmd/plugin/plugin.go [hunk 2]
```diff
@@ -230,7 +226,10 @@ func isExecutable(fullPath string) (bool, error) {
 	}
 
 	if runtime.GOOS == "windows" {
-		if strings.HasSuffix(info.Name(), ".exe") {
+		fileExt := strings.ToLower(filepath.Ext(fullPath))
+
+		switch fileExt {
+		case ".bat", ".cmd", ".com", ".exe":
```
Comment (tedyu) [inline pkg/kubectl/cmd/plugin/plugin.go]: What about the following file extensions ?

lnk -> File Shortcut
bin -> Binary Executable
ps1 -> Windows PowerShell Cmdlet
sct -> Windows Scriptlet
vb , vbs -> VBScript File

If some of the above extensions should be added, I can send out a PR.

Thanks

Comment (neolit123) [reply pkg/kubectl/cmd/plugin/plugin.go]: i don't think some of these will integrate well with the plugin idea.
i'm +1 for ps1

> lnk -> File Shortcut

link files are mostly about GUI shortcuts

> bin -> Binary Executable

it's a binary file of a generic type. different tools have different interpretation of the contents.

> ps1 -> Windows PowerShell Cmdlet

this one is nice to have.

> sct -> Windows Scriptlet
> vb , vbs -> VBScript File

i think we can add these only if users demand them.

Comment (juanvallejo) [reply pkg/kubectl/cmd/plugin/plugin.go]: > i think we can add these only if users demand them.

I'd be in favor of having this be the criteria overall for adding any additional extensions

**Pull request merged**